pkg/bastion/acl.go

```diff
@@ -46,5 +50,51 @@ func checkACLs(user dbmodels.User, host dbmodels.Host) string {
 	}
 	sort.Sort(byWeight(acls))
 
-	return acls[0].Action
+	return checkACLsHook(aclCheckCmd, acls[0].Action, user, host)
+}
+
+// checkACLsHook executes external command to check ACL and passes following parameters:
+// $1 - SSH Portal action (`allow` or `deny`)
+// $2 - User as JSON string
+// $3 - Host as JSON string
+// External program has to return `allow` or `deny` in stdout.
+func checkACLsHook(aclCheckCmd string, action string, user dbmodels.User, host dbmodels.Host) string {
```

Comment:
@Dzalevski I use `string` as it used in original function:
```
func checkACLs(user dbmodels.User, host dbmodels.Host) string {
```
https://github.com/moul/sshportal/pull/212/files#diff-44b7bf87044819f5597a0b7c1fed5a7392402ac70c9976e125092105e93536aaL16

I have to return string `allow` or `deny`.

Here's simple logic:

- if external command executed without errors - return its result (string `allow|deny`)
- if there's any errors, ignore them (print to log only) and return default action

So in any case standard SSH Portal logic will work.

Reply:
if we return the actual erorr, we can just log it on the higher level but it will be a breaking change for that func let's see what @moul will say about this.

Reply:
I also think that it would be better to return an error on the `checkACLs` func

If you feel confident to update the codebase accordingly, feel free to update the function signature; anyway, since `checkACLsHook` is an helper of `checkACLs` and is not called by anything else, I suggest to at least make `checkACLsHook` returning an error to reduce the amount of `log.Printf("Error")` and keep them only in the `checkACLs` func

Reply:
probably fixed :)